components/modal/modal.component.ts

```diff
@@ -76,8 +75,7 @@ export class ModalDirective implements AfterViewInit, OnDestroy {
 
   @HostListener('click', ['$event'])
   protected onClick(event:any):void {
-    if (this.config.backdrop !== 'static' &&
-        event.target === this.element.nativeElement) {
+    if (this.config.ignoreBackdropClick || this.config.backdrop === 'static') {
```

Comment:
@pradeepsekar seem this logic should be reversed?
if ignoreBackdropClick is true or this.config.backdrop === 'static' modal should not be hidden